src/atomWithLocation.ts

```diff
@@ -70,7 +70,7 @@ export function atomWithLocation<T>(options?: Options<T>) {
   baseAtom.onMount = (set) => {
     const callback = () => set(getL());
     const unsub = sub(callback);
-    callback();
+    // callback();
```

Comment:
But, without it, there's a case that it misses an update.
To avoid double updates at init, you need to compare if it's not really changed. It was easy if it's just a string or something.

Reply:
src/atomWithLocation.ts: 
69: const baseAtom = atom(options?.preloaded ?? getL());
The baseAtom had default value at init.

Reply:
if .onMount can be invoked with delay，wo need use **setLocation** to change location。

Reply:
what is setLocation?

I just noticed, to make it consistent with `atomWithHash` #13, L69 should just be `atom(options?.preloaded)` meaning initially it's `undefined`.

Reply:
const locationAtom = atomWithLocation();
const [location, **setLocation**] = useAtom(locationAtom);

The locationAtom was init/setLocation before it's first used, so it isn‘t needed to be setted at onMount.

Reply:
I'm not sure if I follow.
Without onMount init, you will see wrong `location` in an edge case. It's really an edge case, so you won't observe it easily.

Reply:
When many atoms depend locationAtom, i need to compare location every time. 
Or maybe has other better solution?

Reply:
I assume the double updates at the initialization is only about performance concern. Do you have any real issues?